openingd/dualopend.c

```diff
@@ -1035,6 +1035,70 @@ static void report_channel_hsmd(const struct state *state,
 			      tal_hex(tmpctx, msg));
 }
 
+static u8 *msg_for_remote_commit(const tal_t *ctx,
+				 const struct state *state,
+				 struct penalty_base **pbase,
+				 char **error)
+{
+	struct wally_tx_output *direct_outputs[NUM_SIDES];
+	struct bitcoin_tx *remote_commit;
+	struct bitcoin_signature local_sig;
+	struct simple_htlc **htlcs = tal_arr(tmpctx, struct simple_htlc *, 0);
+	u32 feerate = 0; // unused since there are no htlcs
+	u64 commit_num = 0;
+        u8 *msg;
+
+	/* Create commitment tx signatures for remote */
+	remote_commit = initial_channel_tx(NULL, NULL, state->channel,
+					   &state->first_per_commitment_point[REMOTE],
+					   REMOTE, direct_outputs, error);
+
+	if (!remote_commit) {
+		return NULL;
```

Comment:
Normal practice would be error to be tallocated off ctx, not NULL.  You can tal_steal(ctx, error), or just use ctx as first arg to initial_channel_tx....

Reply:
Fwiw none of the strings inside the `inital_channel_tx` call chain are allocated, they're all statically defined.